### PR TITLE
Fixed up Promotions and added claims to webserver

### PR DESCRIPTION
**Pull request by jmazouri**
These changes may seem somewhat disparate, but they are actually related enough to justify making them in one PR.

Promotions:
Fixed various issues with promotion DB storage/retrieval
Fixed issues with mixed scopes in DI causing exceptions
Switched Promotions to use Claims system, removed some previously used config values

Webserver:
Switched from old rank-based system to Claims system, integrated with Modix claims
Removed unused/redundant auth attributes
Enabled data protection disk persistence
Enabled response compression (gzip)
Added "/invite" redirect (goes to aka.ms link)

Web Frontend:
Updated all packages, targeting latest vue-cli version to fix builds
Updated promotion service/models for new backend implementation
Updated User model for new claims system
Other small fixes and typescript-related adjustments

### CODE REVIEW

File: Modix.Services/Core/AuthorizationService.cs [hunk 1]
```diff
@@ -238,12 +238,14 @@ public void RequireAuthenticatedUser()
...
             if (CurrentUserId == null)
                 // TODO: Booooo for exception-based flow control
-                throw new InvalidOperationException("The current operation requires an authenticated guild.");
+                throw new InvalidOperationException("The current operation requires an authenticated user.");
         }
 
         /// <inheritdoc />
         public void RequireClaims(params AuthorizationClaim[] claims)
         {
+            RequireAuthenticatedUser();
```
Comment (JakenVeina) [inline Modix.Services/Core/AuthorizationService.cs]: I mean, I guess there's no scenario where we could ever load claims without a user, but..... I don't really see the need for this as a requirement. /shrug

File: Modix/ModixBot.cs [hunk 2]
```diff
@@ -124,13 +124,13 @@ public async Task HandleCommand(SocketMessage messageParam)
             {
                 var context = new CommandContext(_client, message);
 
+                if (!(context.User is IGuildUser)){ return; }
```
Comment (JakenVeina) [inline Modix/ModixBot.cs]: We should plan to support non-guild users eventually. There are some commands, like help, that we can serve up in DM channels. I forget if it was rowboat or MODiX or another bot that already does this.

File: Modix.Services/Core/AuthorizationService.cs (hunk 1 above)
Comment (jmazouri) [reply Modix.Services/Core/AuthorizationService.cs]: Just a consistency thing, mostly. Without it you get a mysterious "required parameter 'second'" exception on the call to `Except()` if you haven't already called OnAuthenticatedAsync.

File: Modix/ModixBot.cs (hunk 2 above)
Comment (jmazouri) [reply Modix/ModixBot.cs]: Agreed. I put it in for now as many parts of Modix expect a guild context, if I recall.

File: Modix.Services/Core/AuthorizationService.cs (hunk 1 above)
Comment (JakenVeina) [reply Modix.Services/Core/AuthorizationService.cs]: ...wut?

Comment (jmazouri) [reply Modix.Services/Core/AuthorizationService.cs]: I know! It was very confusing:
image

**Pull request merged**